src/net.cpp

```diff
@@ -746,43 +764,70 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
     nLastRecv = nTimeMicros / 1000000;
     nRecvBytes += nBytes;
     while (nBytes > 0) {
-
         // get current incomplete message, or create a new one
-        if (vRecvMsg.empty() ||
-            vRecvMsg.back().complete())
-            vRecvMsg.push_back(CNetMessage(Params().MessageStart(), SER_NETWORK, INIT_PROTO_VERSION));
-
-        CNetMessage& msg = vRecvMsg.back();
+        if (vRecvMsg.empty() || vRecvMsg.back()->complete()) {
+            if (m_encryption_handler && m_encryption_handler->shouldCryptMsg()) {
+                vRecvMsg.emplace_back(std::make_shared<NetCryptedMessageEnvelop>(m_encryption_handler, Params().MessageStart(), SER_NETWORK, INIT_PROTO_VERSION));
```

Comment:
Nit: Should this be ”envelope” instead of ”envelop”? If so, change applies throughout this PR and also the BIP.